### ISSUE TITLE
redefined-outer-name is very commonly emitted when using the TYPE_CHECKING feature

### ISSUE THREAD

**Issue by NeilGirdhar** [PyCQA/pylint#2834]
This is a common pattern when annotating using types that cannot be imported:
```
from __future__ import annotations
from typing import TYPE_CHECKING


class X:

    def f(self, s: Signal):
        # This import makes the definition work.
        from .signal import Signal
        t = Signal("blah")


if TYPE_CHECKING:
    # This import makes the annotations work.
    from .signal import Signal
```

Unfortunately, this gives

```
Redefining name 'Signal' from outer scope (line 10) (redefined-outer-name)
```

The import may not be possible to do at the top of the file because of circular imports.  Perhaps two imports that are from the same place should not produce this error.

**Comment by PCManticore** [comment]
Thank you @NeilGirdhar This makes sense.
